app/main.py

```diff
@@ -1,2 +1,50 @@
+import dataclasses
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.size: int = 0
+        self.capacity: int = 8
+        self.node = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Hashable, value: Any) -> Any:
+        index = hash(key) % self.capacity
+        treshold = self.capacity * 2 / 3
+        while self.node[index] is not None:
+            if self.node[index].key == key:
+                self.node[index].value = value
+                return
+            index = (index + 1) % self.capacity
+        if self.size > treshold:
+            self.resize_hashtable()
+            self.__setitem__(key=key, value=value)
+            return
+        self.size += 1
+        self.node[index] = Node(key=key, value=value, hash_=hash(key))
+
+    def resize_hashtable(self) -> None:
+        self.capacity *= 2
+        items = [item for item in self.node if item is not None]
```

Comment:
use filter + list

Reply:
`use filter + list` what does it mean?

Reply:
```suggestion
        items = list(filter(lambda item: item is not None,self.node))
```

Reply:
but, for my opinion,  it is not necessary